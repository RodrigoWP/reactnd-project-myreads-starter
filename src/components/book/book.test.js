import React from 'react'
import { shallow, mount } from 'enzyme'

import Book, { bookImage } from './index'

const book = {
  imageLinks: {
    thumbnail: ''
  },
  title: '',
  subtitle: ''
}

it('Renderiza sem erros', () => {
  const bookComponent = (
    <Book
      book={book}
      updateShelfBook={() => {}}
    />
  )

  expect(shallow(bookComponent)).toMatchSnapshot()
})

it('Ao chamar bookImage e a imagem for null o caminho deve retornar /no-image-found', () => {
  const imagePath = bookImage(null)

  const imageNotFoundPath = '/no-image-found.jpg'

  expect(imagePath).toBe(imageNotFoundPath)
})

it('Deve chamar updateShelfBook ao dar change no select', () => {
  const updateShelfBook = jest.fn()

  const bookComponent = mount(
    <Book
      book={book}
      updateShelfBook={updateShelfBook}
    />
  )

  bookComponent.find("select").simulate('change')
  expect(updateShelfBook).toHaveBeenCalledTimes(1)
})
