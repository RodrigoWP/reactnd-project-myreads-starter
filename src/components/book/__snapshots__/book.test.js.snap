exports[`test Renderiza sem erros 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="book">
    <div
        className="book-top">
        <div
            className="book-cover"
            style={
                Object {
                    "backgroundImage": "url()",
                  }
            } />
        <div
            className="book-shelf-changer">
            <select
                onChange={[Function]}
                value="none">
                <option
                    disabled={true}
                    value="none">
                    Move to...
                </option>
                <option
                    value="currentlyReading">
                    Currently Reading
                </option>
                <option
                    value="wantToRead">
                    Want to Read
                </option>
                <option
                    value="read">
                    Read
                </option>
                <option
                    value="none">
                    None
                </option>
            </select>
        </div>
    </div>
    <div
        className="book-title" />
    <div
        className="book-authors" />
</div>,
  "nodes": Array [
    <div
      className="book">
      <div
            className="book-top">
            <div
                  className="book-cover"
                  style={
                        Object {
                              "backgroundImage": "url()",
                            }
                  } />
            <div
                  className="book-shelf-changer">
                  <select
                        onChange={[Function]}
                        value="none">
                        <option
                              disabled={true}
                              value="none">
                              Move to...
                        </option>
                        <option
                              value="currentlyReading">
                              Currently Reading
                        </option>
                        <option
                              value="wantToRead">
                              Want to Read
                        </option>
                        <option
                              value="read">
                              Read
                        </option>
                        <option
                              value="none">
                              None
                        </option>
                  </select>
            </div>
      </div>
      <div
            className="book-title" />
      <div
            className="book-authors" />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Book
        book={
                Object {
                        "imageLinks": Object {
                          "thumbnail": "",
                        },
                        "subtitle": "",
                        "title": "",
                      }
        }
        updateShelfBook={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "book": Object {
            "imageLinks": Object {
              "thumbnail": "",
            },
            "subtitle": "",
            "title": "",
          },
          "updateShelfBook": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="book">
          <div
                    className="book-top">
                    <div
                              className="book-cover"
                              style={
                                        Object {
                                                  "backgroundImage": "url()",
                                                }
                              } />
                    <div
                              className="book-shelf-changer">
                              <select
                                        onChange={[Function]}
                                        value="none">
                                        <option
                                                  disabled={true}
                                                  value="none">
                                                  Move to...
                                        </option>
                                        <option
                                                  value="currentlyReading">
                                                  Currently Reading
                                        </option>
                                        <option
                                                  value="wantToRead">
                                                  Want to Read
                                        </option>
                                        <option
                                                  value="read">
                                                  Read
                                        </option>
                                        <option
                                                  value="none">
                                                  None
                                        </option>
                              </select>
                    </div>
          </div>
          <div
                    className="book-title" />
          <div
                    className="book-authors" />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="book">
          <div
                    className="book-top">
                    <div
                              className="book-cover"
                              style={
                                        Object {
                                                  "backgroundImage": "url()",
                                                }
                              } />
                    <div
                              className="book-shelf-changer">
                              <select
                                        onChange={[Function]}
                                        value="none">
                                        <option
                                                  disabled={true}
                                                  value="none">
                                                  Move to...
                                        </option>
                                        <option
                                                  value="currentlyReading">
                                                  Currently Reading
                                        </option>
                                        <option
                                                  value="wantToRead">
                                                  Want to Read
                                        </option>
                                        <option
                                                  value="read">
                                                  Read
                                        </option>
                                        <option
                                                  value="none">
                                                  None
                                        </option>
                              </select>
                    </div>
          </div>
          <div
                    className="book-title" />
          <div
                    className="book-authors" />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Book
    book={
        Object {
            "imageLinks": Object {
              "thumbnail": "",
            },
            "subtitle": "",
            "title": "",
          }
    }
    updateShelfBook={[Function]} />,
}
`;
